packages/synapse-react-client/src/components/SynapseTable/SendToCavaticaConfirmationDialog.tsx

```diff
@@ -46,12 +45,18 @@ export default function SendToCavaticaConfirmationDialog(
     if (!hasSelectedRows) {
       return request
     } else {
+      if (!rowSelectionPrimaryKey || rowSelectionPrimaryKey.length !== 1) {
+        // TODO: Handle composite/undefined key
+        throw new Error(
+          'rowSelectionPrimaryKey must be defined and have length 1',
+        )
+      }
       // Add a filter that will just return the selected rows.
       const idColIndex = data?.columnModels?.findIndex(cm => cm.name === 'id')
       const idColumnFilter: ColumnSingleValueQueryFilter = {
         concreteType:
           'org.sagebionetworks.repo.model.table.ColumnSingleValueQueryFilter',
-        columnName: 'id',
+        columnName: rowSelectionPrimaryKey[0],
         operator: ColumnSingleValueFilterOperator.IN,
         values: selectedRows!.map(row => row.values[idColIndex!]!),
       }
```

Comment:
This entire section should probably get refactored into its own utility (which the cohort builder will use). Perhaps we can figure out a more graceful way to handle the error case then.